sig-robotics/proposal/Generating Action Sequences for Robot Task Plans using LLM/README.md

```diff
@@ -0,0 +1,167 @@
+
+This project use **prompt technique** to drive complex Robot operations, user utilize a prompt like  `Go to the kitchen and help me get an apple.` This prompt makes Robot go to the kitchen and find the apple, take it away, and finally take it back to the owner.
+
+Current research：
+- google [SayCan](https://say-can.github.io/)
+- Microsoft [PromptCraft-Robotics](https://github.com/microsoft/PromptCraft-Robotics/tree/main)
+- Nvidia [VOYAGER](https://github.com/MineDojo/Voyager)
+
+This proposal provides a basic robot LLM Agent (RoboAgent) for RoboSDK, enabling robot developers and users to use LLM prompt to drive robots in an intelligent manner.
+
+# 2. Goals
+For developers or end users of RoboSDK, the goals of the RoboAgent are:
+
+1. Test the perception, planning, and control algorithms of a robot in complex navigation and grasping tasks, both indoors and outdoors, based on given instructions (e.g. "Go to the front of that BYD car and check if anything important has been left behind"). Utilize large language models like ChatGPT for this purpose.
+
+2. Perform closed-loop verification of the developed algorithms in **Gazebo** simulation environment.
+
+3. Deploy the data transmission model and other development processes using RoboSDK.
+
+
+# 3. Proposals
+By building RoboAgent and its related components on RoboSDK, RoboSDK can have the ability to handle complex robot tasks based on LLM (Large Language Models).
+
+The scope of RoboAgent and its related suite includes:
+- Building a RoboAgent capable of task planning, environment observation, and phenomena reasoning based on prompt techniques such as ReAct, self-ask, tree-of-thoughts, to enable complex task planning and processing abilities.
+- Providing a generic interface for large language models (LLM) to empower RoboAgent with inference and planning capabilities.
+- Building the RoboToolKit suite based on prompt techniques like ReAct, which interfaces with the control interface built on RoboSDK to provide action-related instructions and control capabilities to RoboAgent.
+- Providing the ability for RoboAgent to perceive the environment by accessing the sensor interface data of RoboSDK.
+- Providing relevant documentation, sample templates, and auxiliary tools to reduce the learning curve for users.
+- Creating related test cases and test reports.
+
+This design targets developers aiming to build complex robots based on LLM using RoboSDK.
+
+# 4. Design Details
+
+## Architecture Design
+To better build the capabilities of RoboAgent in task planning, environment perception, and reasoning, and to enable RoboAgent to handle complex robot instructions, we need to build the RoboAgent and related tool suites, including Planning, Action, Reasoning, and Observation. The following diagram illustrates the system architecture built based on this concept.
```

Comment:
Ensuring reliable and safe execution of tasks should be a top priority.Maybe you need to consider the "hallucinations" that generating models can bring.